src/NuGet.Core/NuGet.Commands/NuGetLockFileBuilder.cs

```diff
@@ -0,0 +1,102 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using NuGet.Common;
+using NuGet.Frameworks;
+using NuGet.LibraryModel;
+using NuGet.Packaging.Core;
+using NuGet.ProjectModel;
+using NuGet.Shared;
+
+namespace NuGet.Commands
+{
+    public class NuGetLockFileBuilder
+    {
+        private readonly int _lockFileVersion;
+        public NuGetLockFileBuilder(int lockFileVersion)
+        {
+            _lockFileVersion = lockFileVersion;
+        }
+
+        public NuGetLockFile CreateNuGetLockFile(LockFile assetsFile)
+        {
+            var lockFile = new NuGetLockFile()
+            {
+                Version = _lockFileVersion
+            };
+
+            var libraryLookup = assetsFile.Libraries.Where(e => e.Type == LibraryType.Package)
+                .ToDictionary(e => new PackageIdentity(e.Name, e.Version));
+
+            foreach (var target in assetsFile.Targets)
+            {
+                var nuGettarget = new NuGetLockFileTarget()
+                {
+                    TargetFramework = target.TargetFramework,
+                    RuntimeIdentifier = target.RuntimeIdentifier
+                };
+
+                var framework = assetsFile.PackageSpec.TargetFrameworks.FirstOrDefault(
+                    f => EqualityUtility.EqualsWithNullCheck(f.FrameworkName, target.TargetFramework));
+
+                var lilbraries = target.Libraries;
+
+                // check if this is RID-based graph then only add those libraries which differ from original TFM.
+                if (!string.IsNullOrEmpty(target.RuntimeIdentifier))
+                {
+                    var onlyTFM = assetsFile.Targets.First(t => EqualityUtility.EqualsWithNullCheck(t.TargetFramework, target.TargetFramework));
+
+                    lilbraries = target.Libraries.Where(lib => !onlyTFM.Libraries.Any(tfmLib => tfmLib.Equals(lib))).ToList();
+                }
+
+                foreach (var library in lilbraries.Where(e => e.Type == LibraryType.Package))
+                {
+                    var identity = new PackageIdentity(library.Name, library.Version);
+
+                    var dependency = new LockFileDependency()
+                    {
+                        Id = library.Name,
+                        ResolvedVersion = library.Version,
+                        Sha512 = libraryLookup[identity].Sha512,
+                        Dependencies = library.Dependencies
+                    };
+
+                    var framework_dep = framework.Dependencies.FirstOrDefault(
```

Comment:
framework cannot be null since we can't have a framework inside lock file targets which doesn't exists in packageSpec. But if we still want to be safeguard then i can handle that as well.

And agree with naming convention thing, but personally i dont care too much about block level local variables, rather be more concerned about class or method level variables. But I'll change this.

Reply:
If it cannot be null then use `First` which will throw an exception if the value doesn't exist.